fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/DynamicRegistryEvents.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.registry;
+
+import java.util.Collection;
+import java.util.List;
+
+import com.mojang.serialization.Codec;
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.registry.Registry;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.registry.RegistryLoader;
+import net.minecraft.util.Identifier;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.fabricmc.fabric.impl.registry.sync.DynamicRegistryRegistrationContextImpl;
+
+/**
+ * Events relating to dynamic registries.
+ */
+public final class DynamicRegistryEvents {
+	// Events
+
+	/**
+	 * An event for registering new dynamic registries.
+	 */
+	public static final Event<RegisterRegistries> REGISTER_REGISTRIES = EventFactory.createArrayBacked(RegisterRegistries.class, callbacks -> context -> {
+		for (RegisterRegistries callback : callbacks) {
+			callback.registerDynamicRegistries(context);
+		}
+	});
+
+	// TODO: is this good to have?
```

Comment:
IMO no, this still depends on the outside functional interface existing.

Reply:
Yeah, I thought it could be nice if we also had #2111 here but I think I'll make it its own PR.